src/pages/MyPatches.tsx

```diff
@@ -1,36 +1,131 @@
-import React from "react";
-import { PageWrapper } from "components/styles";
+import React, { useState, useEffect } from "react";
+import {
+  PageWrapper,
+  StyledInput,
+  FiltersWrapperSpaceBetween,
+  SimplePageTitle,
+} from "components/styles";
 import { useLocation, useHistory } from "react-router-dom";
 import queryString from "query-string";
 import Checkbox from "@leafygreen-ui/checkbox";
-import { MyPatchesQueryParams } from "types/patch";
-import { H2 } from "@leafygreen-ui/typography";
+import { MyPatchesQueryParams, ALL_PATCH_STATUS } from "types/patch";
+import Icon from "@leafygreen-ui/icon";
+import {
+  UserPatchesVars,
+  UserPatchesData,
+  GET_USER_PATCHES,
+} from "gql/queries/my-patches";
+import { StatusSelector } from "pages/my-patches/StatusSelector";
+import { useQuery } from "@apollo/react-hooks";
+import { useFilterInputChangeHandler } from "hooks";
+import styled from "@emotion/styled";
 
 export const MyPatches = () => {
+  const { replace, listen } = useHistory();
   const { search, pathname } = useLocation();
-  const { replace } = useHistory();
-  const parsed = queryString.parse(search);
-  const showCommitQueue =
-    parsed.commitQueue === "true" || parsed.commitQueue === undefined;
+  const [initialQueryVariables] = useState<UserPatchesVars>({
+    $page: 0,
+    ...getQueryVariables(search),
+  });
+  const [
+    patchNameFilterValue,
+    patchNameFilterValueOnChange,
+  ] = useFilterInputChangeHandler(MyPatchesQueryParams.PatchName);
+  const { data, fetchMore, networkStatus, error } = useQuery<
+    UserPatchesData,
+    UserPatchesVars
+  >(GET_USER_PATCHES, {
+    variables: initialQueryVariables,
+    notifyOnNetworkStatusChange: true,
+  });
+
+  useEffect(() => {
+    return listen(async (loc) => {
+      try {
+        await fetchMore({
+          variables: {
+            pageNum: 0,
+            ...getQueryVariables(loc.search),
+          },
+          updateQuery: (
+            prev: UserPatchesData,
+            { fetchMoreResult }: { fetchMoreResult: UserPatchesData }
+          ) => {
+            if (!fetchMoreResult) {
+              return prev;
+            }
+            return fetchMoreResult;
+          },
+        });
+      } catch (e) {
+        // empty block
+      }
+    });
+  }, [networkStatus, error, fetchMore, listen]);
```

Comment:
this is third appearance of this code that is almost identical to Patch and Task page fetchMore handler func. not a blocker but this would do well as a shared function or hook between the three pages.